lib/jetstream/api/consumer.ex

```diff
@@ -299,15 +302,15 @@ defmodule Jetstream.API.Consumer do
       iex> {:error, %{"code" => 404, "description" => "stream not found"}} = Jetstream.API.Consumer.list(:gnat, "wrong_stream")
 
   """
-  @spec list(conn :: Gnat.t(), stream_name :: binary(), params :: [offset: non_neg_integer()]) ::
+  @spec list(conn :: Gnat.t(), stream_name :: binary(), domain :: nil | binary(), params :: [offset: non_neg_integer()]) ::
           {:ok, consumers()} | {:error, term()}
-  def list(conn, stream_name, params \\ []) do
+  def list(conn, stream_name, domain \\ nil, params \\ []) do
```

Comment:
This is one of the areas where it looks like we needed to update tests and I think making this change would trigger us to release as a new major version since someone who was previously calling the function with 3 arguments will need to fix their callsites.

Do you think it would make sense to make `domain` be a supported option to the `params` list? So people could call the function with `Consumer.list(conn, "MyStream", domain: "leaf_node")`? That strikes me as a more friendly migration path

Reply:
Yes, that is a good suggestion, not changing the API would be best.